openedx/core/djangoapps/content_staging/views.py

```diff
@@ -89,18 +91,31 @@ def post(self, request):
         if usage_key.block_type in ('course', 'chapter', 'sequential'):
             raise ValidationError('Requested XBlock tree is too large')
         course_key = usage_key.context_key
-        if not isinstance(course_key, CourseLocator):
-            # In the future, we'll support libraries too but for now we don't.
-            raise ValidationError('Invalid usage key: not a modulestore course')
-        # Make sure the user has permission on that course
-        if not has_studio_read_access(request.user, course_key):
-            raise PermissionDenied("You must be a member of the course team in Studio to export OLX using this API.")
 
         # Load the block and copy it to the user's clipboard
         try:
-            block = modulestore().get_item(usage_key)
+            if isinstance(course_key, CourseLocator):
+                # Make sure the user has permission on that course
+                if not has_studio_read_access(request.user, course_key):
+                    raise PermissionDenied(
+                        "You must be a member of the course team in Studio to export OLX using this API."
+                    )
+                block = modulestore().get_item(usage_key)
+
+            elif isinstance(course_key, LibraryLocatorV2):
+                lib_api.require_permission_for_library_key(
+                    usage_key,
+                    request.user,
+                    lib_api.permissions.CAN_VIEW_THIS_CONTENT_LIBRARY
+                )
+                block = xblock_api.load_block(usage_key, user=None)
```

Comment:
This wasn't working for me when trying to copy library blocks to the clipboard. It would fail because the [require_permission_for_library_key](https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/content_libraries/api.py#L332-L344) expects the first argument to be a `LibraryLocatorV2` since it calls the [get_by_key](https://github.com/openedx/edx-platform/blob/master/openedx/core/djangoapps/content_libraries/models.py#L75-L79) method internally, and the assertion fails because we are passing in `usage_key`, rather than `course_key` which we confirm is an instance of `LibraryLocatorV2` in the check.

So I needed to change it to the following for it to work:

```suggestion
            elif isinstance(course_key, LibraryLocatorV2):
                lib_api.require_permission_for_library_key(
                    course_key,
                    request.user,
                    lib_api.permissions.CAN_VIEW_THIS_CONTENT_LIBRARY
                )
                block = xblock_api.load_block(usage_key, user=None)
```